File: tests/fixtures/source-windows/ruby/module-class.rb
module Windows
  # Keep report entries in insertion order.
  class ReportBook
    def initialize
      @entries = []
    end

    def add(title, owner)
      @entries << { title: title, owner: owner }
    end

    def render
      @entries.map do |entry|
        owner = entry[:owner] || "unassigned"
        "#{entry[:title]} — #{owner}"
      end
    end

    def size
      @entries.length
    end
  end
end

File: tests/fixtures/source-windows/ocaml/module-class.ml
module Report = struct
  type entry = {
    title : string;
    owner : string option;
  }

  class report_book = object
    val mutable entries : entry list = []

    method add title owner =
      entries <- entries @ [{ title; owner }]

    method render =
      List.map
        (fun entry ->
          let owner = Option.value entry.owner ~default:"unassigned" in
          entry.title ^ " — " ^ owner)
        entries

    method size = List.length entries
  end
end
